monitor.py

```diff
@@ -680,8 +680,12 @@ def where_clause(self):
         return self.timestamp_field < self.cutoff
 
     def run_once(self, *args, **kwargs):
-        rows_deleted = self.query().delete(synchronize_session=False)
-        self.log.info("Deleted %d row(s)", rows_deleted)
+        rows_deleted = 0
+        qu = self.query()
+        self.log.info("Deleting %d row(s)", qu.count())
+        for i in qu:
```

Comment:
This is the actual change; iterating over the query results instead of doing one big delete statement.

Reply:
Is this faster? or why?

Reply:
Never mind, I get it now.

Reply:
It's slower, because SQLAlchemy will use separate SQL statements to cascade the delete to every other object associated with this object (e.g. all of this patron's loans and credentials). 

Trying to delete all the patrons at once is faster, and it would work if we did the cascades on a database level, but it doesn't work here.